### PR TITLE
[JUJU-4399] Remove JujuConnSuite entirely and slim down dummy provider

### PR DESCRIPTION
**Pull request by wallyworld**
TL;DR; remove JujuConnSuite and gut the dummy provider to remove database and state concerns

The JujuConnSuite test fixture and dummy provider formed the basis for most of the end to end test suites. They were heavy weight and the separation of concerns between the two was blurred, eg the dummy provider set up persistent state which is not something a provider is supposed to care about. It meant the transition away from mongo would be impractical if this infrastructure remained.

This PR replaces JujuConnSuite with the previously introduced ApiServer test fixture - this fixture is simply an api server over a database; out of the box only a minimal server is created; extras like all watcher, lease manager etc are opt in. We also move away from obsolete state helpers to instead create model artefacts using the preferred test factory.

The dummy provider is still present, but it now is just a skeleton Environ. The (much reduced) places it is imported do not use it directly but do so just to have it registered as a provider so that tests which need a cloud and associated environ will work. Further refactoring can rip out even more code, eg the bootstrap CLI tests can be improved to use mocks and avoid the dummy provider entirely.

It turns out some jujud tests failed to stub out calls real executables in various workers, eg lsblk in the storage worker. This has been fixed, now that the ApiServer fixture enforces  a clean execution environment.

## QA steps

run unit tests

### CODE REVIEW

File: doc/how-to-write-tests.txt [hunk 1]
```diff
@@ -163,11 +163,11 @@ can instead embedd one of the fake home suites:
 * FakeHomeSuite: creates a fake home directory with ~/.ssh and fake ssh keys.
 * FakeJujuHomeSuite: as above but also sets up a ~/.config/juju with a fake model.
 
-The JujuConnSuite does this and more. It also sets up a controller and api
-server.  This is one problem with the JujuConnSuite, it almost always does a
+The ApiServerSuite does this and more. It also sets up a controller and api
```
Comment (hpidcock) [inline doc/how-to-write-tests.txt:166]: Is this all true still?

File: doc/contributions/review-checklist.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ A list of common (not exhaustive) mistakes to check for when reviewing. For furt
 - Could similar tests be combined into a table based test?
 - Should a table based test be broken out in more obvious tests? (this sometimes happens when there is too much logic or too many conditionals in the actual test block of the table based test).
 - Is there repeated test boilerplate which should be extracted in functions on the testSuite?
-- Is the appropriate base test suite used? (overuse of JujuConnSuite is common).
+- Is the appropriate base test suite used? (overuse of ApiServerSuite is common).
```
Comment (hpidcock) [inline doc/contributions/review-checklist.md:46]: Is this also true still?

**Review by hpidcock — APPROVED** [review]
Nice, thanks.

File: doc/contributions/review-checklist.md (hunk 2 above)
Comment (wallyworld) [reply doc/contributions/review-checklist.md:46]: yeah, but we're gonna progressively address it as facades come across to dqlite

File: doc/how-to-write-tests.txt (hunk 1 above)
Comment (wallyworld) [reply doc/how-to-write-tests.txt:166]: Arguably it is. It's far less bad than it was. I think once we get more new infrastructure in place, we can revisit this text.
